tpm/constants.go

```diff
@@ -71,6 +72,62 @@ const (
 	tpmCapFlagPermanent    uint32 = 0x00000108
 )
 
+// permission type
+type permission uint32
+
+// NV Permissions and Operations
+// Note: Permissions are summable
+const (
+	nvPerPPWrite    permission = 0x00000001
+	nvPerOwnerWrite permission = 0x00000002
+	nvPerAuthWrite  permission = 0x00000004
+	nvPerWriteAll   permission = 0x00000800
+	// Warning: The Value 0x00001000 is
+	// defined in the spec as
+	// TPM_NV_PER_WRITEDEFINE, but it is
+	// not included directly in this
+	// code because it locks the given
+	// NV Index permanently if used
+	// incorrectly. This operation can't
+	// be undone in any way. Do not use
+	// this value unless you know what
+	// you're doing!
+	nvPerWriteSTClear permission = 0x00002000
+	nvPerGlobalLock   permission = 0x00004000
+	nvPerPPRead       permission = 0x00008000
+	nvPerOwnerRead    permission = 0x00100000
+	nvPerAuthRead     permission = 0x00200000
+	nvPerReadSTClear  permission = 0x80000000
+)
+
+// PermMap : Map of TPM_NV_Permissions to its strings for convenience
+var PermMap = map[permission]string{
+	nvPerPPWrite:      "PPWrite",
+	nvPerOwnerWrite:   "OwnerWrite",
+	nvPerAuthWrite:    "AuthWrite",
+	nvPerWriteAll:     "WriteAll",
+	nvPerWriteSTClear: " WriteSTClear",
+	nvPerGlobalLock:   "GlobalLock",
+	nvPerPPRead:       "PPRead",
+	nvPerOwnerRead:    "OwnerRead",
+	nvPerAuthRead:     "AuthRead",
+	nvPerReadSTClear:  "ReadSTClear",
+}
+
+// PrettyPrinting for Permissions
+func (p permission) String() string {
+	var retString string
+	for iterator, item := range PermMap {
+		p = p - iterator
+		if p == 0 {
```

Comment:
It seems like what you are trying to do here is test the bits to see if the permission is contained in `p`?

If so, the way you do that is like this:

```go
if (p & iterator) != 0 {
  // permission is set
}
```

And then return `retString` at the end.

Also, consider using `strings.Builder` from the standard library for this - its more idomatic and more efficient.